caddy/caddymain/run.go

```diff
@@ -82,6 +82,7 @@ func Run() {
 	certmagic.UserAgent = appName + "/" + appVersion
 
 	// Set up process log before anything bad happens
+	caddy.LogDestination = logfile
```

Comment:
Instead of setting this here, why not have the flag call in the init function set `caddy.LogDestination` directly, and switch on that?